snapcraft/internal/errors.py

```diff
@@ -168,39 +182,44 @@ def __init__(self, filename, message):
 class SnapcraftPartMissingError(SnapcraftError):
 
     fmt = (
-        'Cannot find the definition for part {part_name!r}.\n'
-        'It may be a remote part, run `snapcraft update` '
-        'to refresh the remote parts cache.'
+        'Failed to get part information: '
+        'Cannot find the definition for part {part_name!r}. '
+        'If it is a remote part, run `snapcraft update` '
+        'to refresh the remote parts cache. '
+        'If it is a local part, make sure that it is defined in the '
+        '`snapcraft.yaml`.'
     )
 
 
 class PartNotInCacheError(SnapcraftError):
 
     fmt = (
-        'Cannot find the part name {part_name!r} in the cache. Please '
-        'run `snapcraft update` and try again.\nIf it is indeed missing, '
-        'consider going to https://wiki.ubuntu.com/snapcraft/parts '
-        'to add it.'
+        'Failed to get remote part information: '
+        'Cannot find the part name {part_name!r} in the cache. '
+        'If it is an existent remote part, run `snapcraft update` and try '
+        'again. If it has not been defined, consider going to '
+        'https://wiki.ubuntu.com/snapcraft/parts to add it.'
     )
 
 
 class PluginError(SnapcraftError):
 
-    fmt = 'Issue while loading part: {message}'
+    fmt = (
+        'Failed to load plugin: '
+        '{message}'
+        # FIXME include how to fix each of the possible plugin errors.
+        # https://bugs.launchpad.net/snapcraft/+bug/1727484
+        # --elopio - 2017-10-25
+    )
 
     def __init__(self, message):
         super().__init__(message=message)
 
 
-class PluginNotDefinedError(SnapcraftError):
-
-    fmt = ("Issues while validating snapcraft.yaml: the 'plugin' keyword is "
-           "missing for the {part_name} part.")
-
-
 class SnapcraftPartConflictError(SnapcraftError):
 
     fmt = (
+        'Failed to run the stage step: '
         'Parts {other_part_name!r} and {part_name!r} have the following file '
         'paths in common which have different contents:\n'
         '{file_paths}\n\n'
```

Comment:
Do you feel like this is more readable?

```
'Failed to stage {part_name!r}: '
'It and {other_part_name!r} both have the following files, but with different contents:[...]'
```

Reply:
Definitely sounds clearer to me and more like something I'd say

Reply:
You are adding too many unnecessary words, it's hard for me to strip them :)
Also, I was kind of trying to keep the first sentence generic, without parameters. Then the second includes parameters to to specify why the problem happened. I'm not sure if that's the best idea, but found that a comfortable consistency. Please review the new version I'm pushing.